sklearn/neural_network/_multilayer_perceptron.py

```diff
@@ -84,9 +84,10 @@ class BaseMultilayerPerceptron(BaseEstimator, metaclass=ABCMeta):
         "beta_1": [Interval(Real, 0, 1, closed="left")],
         "beta_2": [Interval(Real, 0, 1, closed="left")],
         "epsilon": [Interval(Real, 0, None, closed="neither")],
-        # TODO update when ScalarOptions available
-        # [Interval(Integral, 1, None, closed="left"), ScalarOptions({np.inf})]
-        "n_iter_no_change": [Interval(Real, 0, None, closed="right")],
+        "n_iter_no_change": [
+            Interval(Integral, 1, None, closed="left"),
+            Options(Real, {np.inf}),
```

Comment:
This is a rather complex way to deal with closed intervals. What do you think about being strict with `Interval` when it comes to the closed parameter? Specifically:

- `Interval(Real, 0, None, closed="both")` means $0 <= x <= \inf$.
- `Interval(Real, 0, None, closed="left")` means $0 <= x < \inf$.
- `Interval(Real, 0, None, closed="right")` means $0 < x <= \inf$.
- `Interval(Real, 0, None, closed="neither")` means $0 < x < \inf$.

Reply:
I don't really get it here. when `closed="right"`, we already `np.inf` to be a valid entry, isn't it?

Reply:
This is already the existing behavior. The issue is with `Integral` intervals. `inf` can't belong to the interval because it's not an `int`. We could adapt `Interval` to support integers and inf but I'd rather not to because I think constraints should be about a -single- given type.

Reply:
Oh I missed the `Integral` point. So yes, I am fine with this.

Reply:
Ah that makes sense. I'm okay with the current behavior with ints.